src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -729,5 +732,150 @@ protected override void AlignBaseOffsetIfNecessary(MetadataType type, ref Layout
             }
         }
 
+
+        private static HashSet<TypeDesc> _reportedManagedSequential = new HashSet<TypeDesc>();
+
+        public static bool IsManagedSequentialType(TypeDesc type)
+        {
+            ProbeType(type);
+            return IsManagedSequentialTypeInternal(type);
+        }
+
+        public static void ProbeType(TypeDesc type)
+        {
+            if (type != null && _reportedManagedSequential.Add(type))
+            {
+                ReportType(type);
+                ProbeType(type.BaseType);
+                foreach (TypeDesc intf in type.RuntimeInterfaces)
+                {
+                    ProbeType(intf);
+                }
+                foreach (TypeDesc inst in type.Instantiation)
+                {
+                    ProbeType(inst);
+                }
+                foreach (FieldDesc field in type.GetFields())
+                {
+                    ProbeType(field.FieldType);
+                }
+            }
+        }
+
+        private static void ReportType(TypeDesc type)
+        {
+            StringBuilder builder = new StringBuilder();
+            builder.Append("[[[IsManagedSequential{");
+            FormatTypeName(builder, type);
+            builder.Append("}=");
+            builder.Append(IsManagedSequentialTypeInternal(type));
+            builder.Append("]]]");
+            Console.WriteLine(builder.ToString());
+        }
+
+        private static void FormatTypeName(StringBuilder result, TypeDesc type)
```

Comment:
We have an abstract class `TypeNameFormatter` (and a bunch of deriving classes) that do this - it lets you avoid some of the common boilerplate.